wand-watcher.lic

```diff
@@ -0,0 +1,153 @@
+no_kill_all
+no_pause_all
+
+custom_require.call(%w[common common-items])
+
+class WandWatcher
+  include DRC
+  include DRCI
+
+  def initialize
+    settings = get_settings();
+    @wand_list = settings.wands
+    @queued_wands = []
+    @no_use_scripts = settings.wand_watcher_no_use_scripts || []
+    @startup_delay = settings.wand_watcher_startup_delay || 10
+    @passive_delay = settings.wand_watcher_passive_delay || 60
+    @activation_failure_messages = [/^The .* remains inert/]
+    UserVars.wand_watcher_timers = {} unless UserVars.wand_watcher_timers
+
+    #don't allow users to make foolish mistakes, always prevent use during go2/burgle
+    @no_use_scripts << 'go2'
+    @no_use_scripts << 'burgle'
+
+    # rebuild UserVars based on what wands are listed in the yaml
+    # this preserves timers existing wands, removes timers for wands no longer in the yaml
+    # and sets new wand next use timers to now
+    temp_timers = {}
+    @wand_list.each do |key,item|
+      if UserVars.wand_watcher_timers[key]
+        temp_timers[key] = UserVars.wand_watcher_timers[key]
+      else
+        temp_timers[key] = Time.now
+      end
+    end
+    UserVars.wand_watcher_timers = temp_timers
+    
+    #startup delay so you can control spammyness of autostart scripts
+    pause @startup_delay
+
+    #run passively
+    passive_run
+    #TODO: run once option
+  end
+
+  def passive_run
+    loop do
+      if @wand_list == nil || @wand_list.empty?
+        #@wand_list = nil when no wands configured in yaml
+        #@wand_list = empty when all wands are deleted due to problems finding wands
+        message "No wands to check.  Ending script.  Double check your settings."
+        exit
+      end
+      check_timers
+      pause @passive_delay
+    end
+  end
+
+  def check_timers  
+    #if any no_use_scripts are running, loop every 10 seconds checking until not
+    loop do
+      break if !(@no_use_scripts.any? { |name| Script.running?(name) })
+      pause 10
+    end
+    
+    #Loop through the wand timers in the UserVars, and queue up any wands that are ready to go
+    UserVars.wand_watcher_timers.each do |wand,next_use_time|
+      if Time.now > next_use_time
+        @queued_wands << wand
+      end
+    end
+
+    #If you have any queued wands, use them, then reset the list of queued wands
+    if !@queued_wands.empty?
+      use_wands
+      @queued_wands = []
+    end
+  end
+
+  def use_wands
+    #pause scripts to prevent interference
+    scripts_to_unpause = []
+    scripts_to_unpause = smart_pause_all
+    
+    #if both hands are full, store the left hand item in a variable for future use and lower the 
+    #left hand to the feet slot.  
+    #left causes less problems with combat (less likely to be a loaded weapon)
+    #feet slot causes less space/correct container issues
+    temp_left_item = nil
+    if DRC.right_hand && DRC.left_hand
+      temp_left_item = DRC.left_hand
+      bput("lower ground left",/^You lower/)
+    end
```

Comment:
Ingenius! I didn't think of this as a viable option for emptying hands without knowing _where_ they'd end up stored until I saw this. Went and tested and sure enough, more than one item can be in your "at feet" slot, pretty cool!